tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c

```diff
@@ -0,0 +1,523 @@
+	fp_storage_pn_ram_clear();
+	fp_storage_manager_ram_clear();
+}
+
+static void setup_fn(void)
+{
+	int err;
+
+	fp_storage_clear();
+
+	err = settings_load();
+	zassert_ok(err, "Unexpected error in settings load");
+}
+
+static void initial_setup_fn(void)
+{
+	int err;
+
+	err = settings_subsys_init();
+	zassert_ok(err, "Error while initializing settings subsys");
+
+	setup_fn();
+}
+
+static void teardown_fn(void)
+{
+	fp_storage_clear();
+}
+
+static void final_teardown_fn(void)
+{
+	int err;
+
+	teardown_fn();
+
+	/* Non-volatile data of the test must be cleared at exit. */
+	err = settings_delete(SETTINGS_RESET_TEST_FULL_NAME(SETTINGS_STAGE_KEY_NAME));
+	zassert_ok(err, "Error while cleaning up the test non-volatile storage");
+
+	err = settings_delete(SETTINGS_RESET_TEST_FULL_NAME(SETTINGS_FINISHED_MASK_KEY_NAME));
+	zassert_ok(err, "Error while cleaning up the test non-volatile storage");
+}
+
+static void personalized_name_store(const char *name)
+{
+	int err;
+
+	err = fp_storage_pn_save(name);
+	zassert_ok(err, "Unexpected error in name save");
+}
+
+static void personalized_name_validate_loaded(const char *name)
+{
+	int err;
+	char read_name[FP_STORAGE_PN_BUF_LEN];
+
+	err = fp_storage_pn_get(read_name);
+	zassert_ok(err, "Unexpected error in name get");
+	err = strcmp(read_name, name);
+	zassert_ok(err, "Invalid Personalized Name");
+}
+
+static void storage_validate_empty(void)
+{
+	int err;
+	char read_name[FP_STORAGE_PN_BUF_LEN];
+	int read_count;
+	struct fp_account_key account_key_list[ACCOUNT_KEY_MAX_CNT];
+
+	read_count = fp_storage_ak_count();
+	zassert_equal(read_count, 0, "Invalid Account Key count");
+
+	err = fp_storage_ak_get(account_key_list, &read_count);
+	zassert_ok(err, "Unexpected error in Account Key get");
+	zassert_equal(read_count, 0, "Invalid Account Key count");
+
+	err = fp_storage_pn_get(read_name);
+	zassert_ok(err, "Unexpected error in name get");
+	err = strcmp(read_name, "");
+	zassert_ok(err, "Expected saved name to be empty");
+}
+
+static void storage_validate_ram(void)
+{
+	static const uint8_t seed = 5;
+	static const uint8_t key_count = ACCOUNT_KEY_MAX_CNT;
+	static const char *name = "storage_validate_ram";
+	int err;
+
+	storage_validate_empty();
+
+	cu_account_keys_generate_and_store(seed, key_count);
+	personalized_name_store(name);
+
+	cu_account_keys_validate_loaded(seed, key_count);
+	personalized_name_validate_loaded(name);
+
+	fp_storage_ak_ram_clear();
+	fp_storage_pn_ram_clear();
+	err = settings_load();
+	zassert_ok(err, "Unexpected error in settings load");
+
+	cu_account_keys_validate_loaded(seed, key_count);
+	personalized_name_validate_loaded(name);
+}
+
+static void storage_validate_nvs(void)
+{
+	int err;
+
+	storage_validate_empty();
+
+	fp_storage_ak_ram_clear();
+	fp_storage_pn_ram_clear();
+	err = settings_load();
+	zassert_ok(err, "Unexpected error in settings load");
+
+	storage_validate_ram();
+}
+
+static void test_as_finished_mark(enum reset_test_finished bit_pos)
```

Comment:
Maybe we could name it `mark_test_as_finished`?

It is kinda weird to read in my opinion in current form + prefix `test_` is mainly used for test case functions i think

Reply:
I tried to follow the conception of trailing verb but maybe it would be better to not start the function name with `test_` prefix.